src/Microsoft.Framework.Logging.Debug/DebugLogger.cs

```diff
@@ -0,0 +1,77 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections;
+using System.Diagnostics;
+using System.Text;
+
+namespace Microsoft.Framework.Logging.Debug
+{
+    public class DebugLogger : ILogger
+    {
+        private readonly Func<string, LogLevel, bool> _filter;
+        private readonly string _name;
+
+        public DebugLogger(string name)
+            : this(name, filter: null)
+        {
+        }
+
+        public DebugLogger(string name, Func<string, LogLevel, bool> filter)
+        {
+            _name = string.IsNullOrEmpty(name) ? nameof(DebugLogger) : name;
+            _filter = filter ?? ((category, logLevel) => true);
```

Comment:
I think we should just leave `_filter` as `null` and check for it in `IsEnabled` rather than assign a default `true` returning delegate. Why run anything at all when checking `IsEnabled` if we can just check for `null`?